### DISCUSSION TITLE
Bug maybe? Calling `broadcast` even after joining a `realtimeChannel` results in error.

### DISCUSSION THREAD

**Discussion by boulaycote** [supabase-community/supabase-kt#121] [Q&A] [answered]
Hi,

Here's my problem, I'm trying to call `broadcast` but am getting 

```
Error while sending message
W  java.lang.IllegalStateException: Cannot broadcast to a channel you didn't join. Did you forget to call join()?
```

I have already joined the channel and am -receiving- broadcasts but cannot send any.

I have looked in the codebase of RealtimeChannel and see where the `JOINED` status is set, and `event=system` is never received (RealtimeChannel.kt:137), and so `_status.value` is never set to `JOINED`.

Any insights?

**Comment by jan-tennert** [comment]
> Hi,
> 
> Here's my problem, I'm trying to call `broadcast` but am getting
> 
> ```
> Error while sending message
> W  java.lang.IllegalStateException: Cannot broadcast to a channel you didn't join. Did you forget to call join()?
> ```
> 
> I have already joined the channel and am -receiving- broadcasts but cannot send any.
> 
> I have looked in the codebase of RealtimeChannel and see where the `JOINED` status is set, and `event=system` is never received (RealtimeChannel.kt:137), and so `_status.value` is never set to `JOINED`.
> 
> Any insights?

Seems like the `system` event is never sent when not listening for postgres changes. I just published `0.9.4-dev-2` for testing purposes. Can you try it out and tell me if it works?

**Reply by boulaycote** [reply]
Yep this works.

Maybe I should try and listen for any postgres_changes?

**Reply by jan-tennert** [reply]
No, that's not necessary, if you don't need it. As it works now, I'll publish a full 0.9.4 later with this fix included

**Reply by boulaycote** [reply]
Awesome thanks

**Reply by boulaycote** [reply]
This appears to have fixed a bug within `Presence` as well, where I could not get the payload data from the `joins`. 

And now I can.

**Comment by jan-tennert** [comment, marked as answer]
Fixed in 0.9.4
